moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -714,6 +738,267 @@ TEST_F(OneRobot, rigidlyConnectedParent)
+  auto robot_model = std::make_shared<moveit::core::RobotModel>(urdf_model, srdf_model);
+
+  moveit::core::RobotState state(robot_model);
+  const moveit::core::JointModelGroup* jmg = state.getJointModelGroup("base_to_tip");
+
+  // Some made-up numbers, at zero and non-zero robot configurations.
+  CheckJacobian(&state, *jmg, makeVector({ 0.0, 0.0 }), makeVector({ 0.1, 0.4 }));
+  CheckJacobian(&state, *jmg, makeVector({ 0.1, 0.4 }), makeVector({ 0.5, 0.1 }));
+}
+
+TEST(getJacobian, RevolutePlanarAndPrismaticJoints)
+{
+  constexpr char kRobotURDF[] = R"(
+      <?xml version="1.0" ?>
+      <robot name="one_robot">
+        <link name="base_link"/>
+        <joint name="joint_a_planar" type="planar">
+            <axis xyz="0 0 1"/>
+            <parent link="base_link"/>
+            <child link="link_a"/>
+            <origin rpy="0 0 0" xyz="0 0 0"/>
+            <limit effort="100.0" lower="-1.0" upper="1.0" velocity="0.2"/>
+        </joint>
+        <link name="link_a"/>
+        <joint name="joint_b_revolute" type="revolute">
+          <axis xyz="0 0 1"/>
+          <parent link="link_a"/>
+          <child link="link_b"/>
+          <origin rpy="0 0 0" xyz="0.0 0.5 0"/>
+          <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.2"/>
+        </joint>
+        <link name="link_b"/>
+        <joint name="joint_c_prismatic" type="prismatic">
+          <axis xyz="0 1 0"/>
+          <parent link="link_b"/>
+          <child link="link_c"/>
+          <origin rpy="0 0 0" xyz="0.2 0.2 0"/>
+          <limit effort="100.0" lower="-1.0" upper="1.0" velocity="0.2"/>
+        </joint>
+        <link name="link_c"/>
+        <joint name="joint_d_revolute" type="revolute">
+          <axis xyz="1 0 0"/>
+          <parent link="link_c"/>
+          <child link="link_d"/>
+          <origin rpy="0 0 0" xyz="0.0 0.2 0.4"/>
+          <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.2"/>
+        </joint>
+        <link name="link_d"/>
+      </robot>
+    )";
+
+  constexpr char kRobotSRDF[] = R"xml(
+      <?xml version="1.0" ?>
+      <robot name="one_robot">
+        <group name="base_to_tip">
+          <joint name="joint_a_planar"/>
+          <joint name="joint_b_revolute"/>
+          <joint name="joint_c_prismatic"/>
+          <joint name="joint_d_revolute"/>
+        </group>
+      </robot>
+      )xml";
+
+  urdf::ModelInterfaceSharedPtr urdf_model = urdf::parseURDF(kRobotURDF);
+  ASSERT_TRUE(urdf_model);
+  auto srdf_model = std::make_shared<srdf::Model>();
+  ASSERT_TRUE(srdf_model->initString(*urdf_model, kRobotSRDF));
+  auto robot_model = std::make_shared<moveit::core::RobotModel>(urdf_model, srdf_model);
+
+  moveit::core::RobotState state(robot_model);
+  const moveit::core::JointModelGroup* jmg = state.getJointModelGroup("base_to_tip");
+
+  CheckJacobian(&state, *jmg, makeVector({ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 }),
+                makeVector({ 0.2, 0.05, 0.1, 0.2, 0.3, 0.4 }));
```

Comment:
do these velocity values intentionally exceed the specified limits (not that it should matter much for the test implementation)?

Reply:
No, not intentional, but it doesn't really matter for the test.